services/FeefoReviewService.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace Craft;
+
+class FeefoReviewService extends BaseApplicationComponent
+{
+  public $plugin = null;
+  public $settings = array();
+
+  public function __construct()
+  {
+    $this->plugin   = craft()->plugins->getPlugin('feeforeview');
+    $this->settings = $this->plugin->getSettings();
+  }
+
+  public function getReviewSummary($id)
+  {
+    if ($this->feefoMerchantId() == null) return null;
+
+    $products = $this->getProducts();
+    if ($products == null) return null;
+
+    $key = array_search($id, array_column($products, 'vendor_ref'));
+    return $key !== false ? $products[$key] : null;
+  }
+
+  public function feefoMerchantId() {
+    return $this->settings->feefoMerchantId;
+  }
+
+  private function getProducts() {
+    $cacheKey = 'feefo_'.$this->feefoMerchantId();
+    $products = craft()->cache->get($cacheKey);
+    if ($products)
```

Comment:
it is a good practice to be more explicit of what you are testing here since I'm not even sure what actually gets tested when you do this, the equivalent test should be either `!empty($products)` or `!is_null($products)` but I'm not actually sure which one it is

Reply:
I think we can use  !is_null($products)!